### PR TITLE
Fix getMultipleSelectValue for HTMLOptionsCollection

### PR DESCRIPTION
**Pull request by tizz98**
I've been getting the following error when using this library with `react-bootstrap` (https://react-bootstrap.github.io/)

```
options.filter is not a function
```

This is because the ref being passed to `register` is actually `HTMLOptionsCollection` not `HTMLOptionElement[]`. I fixed this temporarily by hacking `HTMLOptionsCollection` but it would be nice not to have to do that.

### CODE REVIEW

File: src/logic/getMultipleSelectValue.ts [hunk 1]
```diff
@@ -1,2 +1,5 @@
-export default (options: HTMLOptionElement[]): string[] =>
+const filterOptions = (options: HTMLOptionElement[]): string[] =>
```
Comment (bluebill1049) [inline src/logic/getMultipleSelectValue.ts]: export default (options: HTMLOptionElement[] | HTMLOptionsCollection): string[] =>	const filterOptions = (options: HTMLOptionElement[]): string[] =>
  [...options].filter(({ selected }): boolean => selected).map(({ value }): string => value);

**Review by bluebill1049 — CHANGES_REQUESTED** [review]

**Comment by bluebill1049** [comment]
thanks very much for the PR ❤️

File: src/logic/getMultipleSelectValue.ts (hunk 1 above)
Comment (bluebill1049) [reply src/logic/getMultipleSelectValue.ts]: I think we can simplify the above to what I suggested :)

Comment (stramel) [reply src/logic/getMultipleSelectValue.ts]: agreed

**Comment by bluebill1049** [comment]
thank you :)

**Comment by bluebill1049** [comment]
@tizz98 I will publish the patch tonight. if you need it now. you can use https://cdn.pika.dev/react-hook-form/master

**Comment by bluebill1049** [comment]
published :) https://github.com/react-hook-form/react-hook-form/releases/tag/v3.22.0

**Pull request merged**